docs/DeveloperGuide.md

```diff
@@ -439,6 +439,11 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 7.  A more technical user must be able to interpret the contents of the stored JSON file.
 8.  A user must be updated on the status of their command within (at most) 2 seconds.
 
+### Planned Enhancements
+
+1. The current phone number validation ensures numbers must be at least 3 digits long and consist of digits (0 to 9). While spaces and dashes are allowed, only 1 space or hyphen is allowed between each digit. This supports common formats like `9143 9032`, `872-233-8554`, and `987223414`.
+   The 3-digit minimum was initially designed to accommodate emergency numbers, but since this is a contact book application, we plan to enhance the phone validation by enforcing a minimum of 7 digits and maximum of 15 digits (excluding formatting characters and area code in parentheses). We will extend the allowed characters to include `+` and parentheses `( )` for global formats. The plus sign must only appear at the start followed by either a digit or open parenthesis, and parentheses must come in matching pairs around area codes (maximum 4 digits). Only one type of separator (space or hyphen) will be allowed, with no spaces inside parentheses. The system will automatically format irregular spacing (e.g., `22 22 22 22` to `2222 2222`) and group numbers by length: 7-8 digits as `XXX-XXXX` or `XXXX-XXXX`, 10 digits as `XXX-XXX-XXXX`, and others in 3s or 4s from right to left. This will accommodate formats like `+1 (234) 567-8901`, `+44 20 7123 4567`, and `+65 6789-0123`, while rejecting invalid patterns like nested parentheses, mixed separators, or multiple plus signs.
```

Comment:
We already allow hyphens, spaces and brackets; but we do not accept `+` yet so that you can keep.

Reply:
Sorry, we don't accept brackets so that can be kept in this paragraph.

Reply:
This pull request may be a bit outdated since we are indeed changing to allow + in phone numbers now? Also, as i sent in the group unrealistic inputs like excessively long phone numbers are not considered bugs as well. I dont think this planned enhancement is needed

Reply:
Yeah I'll close this one